crates/uv/tests/it/init.rs

```diff
@@ -2253,6 +2253,36 @@ fn init_failure() -> Result<()> {
     Ok(())
 }
 
+#[test]
+fn init_failure_with_invalid_option_named_backend() {
+    let context = TestContext::new("3.12");
+
+    uv_snapshot!(context.filters(), context.init().arg("foo").arg("--backend"), @r###"
+    success: false
+    exit_code: 2
+    ----- stdout -----
+
+    ----- stderr -----
+    error: a value is required for '--backend <BACKEND>' but none was supplied
```

Comment:
This seems suboptimal. What happens if we make `--backend` a flag, so it doesn't have a value?

Reply:
I think that is better. I found this error message annoying  but didn't remember the flag alternative. The other alternative I thought about was making `--backend` and alias.

Reply:
Yeah I agree this is suboptimal. A flag might make sense but then what happens if you provide a value? 

Unfortunately this is sort of the status quo upstream, from what I understood in https://github.com/clap-rs/clap/discussions/5781#discussioncomment-11104839

Reply:
I don't really want to make it an alias, it'd be nice to just be able to point people to the canonical name in a way that doesn't cause a bunch of weird behaviors :(

Reply:
Making  this a flag doesn't seem to be an option since it breaks the `build-backend` option.